src/core/CameraControls.tsx

```diff
@@ -41,7 +41,8 @@ export const CameraControls = forwardRef<CameraControlsImpl, CameraControlsProps
   }, -1)
 
   useEffect(() => {
-    return () => void cameraControls.dispose()
+    cameraControls.connect(explDomElement)
+    return () => void cameraControls.disconnect()
```

Comment:
@abernier I think we could keep `dispose()` at cleanup as same as other controls in drei. `disconnect()` may remain some features of CameraControls instance.
What do you think?
```suggestion
    return () => void cameraControls.dispose()
```

Reply:
in my understanding, r3f will take care of calling `.dispose` thanks to https://docs.pmnd.rs/react-three-fiber/api/objects#disposal

Here it is more about "re-connecting" (`disconnect` + `connect`) once any useEffect deps changes

@drcmda or @CodyJasonBennett can you confirm?

Reply:
Connect/disconnect should do and undo effectful stuff like listeners. I'd expect `dispose` to not be reversible, repeatable, or cheap to do at runtime.

Reply:
Thank you @CodyJasonBennett, so in my mind this looks ok to me 👍🏻

Reply:
@abernier 
That makes sense. Thank you for considering it.